### DISCUSSION TITLE
AssertionError in 3.15: "either include a `queryset` attribute, or override the `get_queryset()` method"

### DISCUSSION THREAD

**Discussion by kevinrenskers** [encode/django-rest-framework#9302] [Potential Issue] [answered]
I have a view where users can fetch and patch their own account settings:

``` python
class AccountSettingsView(RetrieveUpdateAPIView):
    serializer_class = AccountSettingsSerializer
    permission_classes = [permissions.IsAuthenticated]

    def get_object(self):
        return AccountSettings.objects.get(user=self.request.user)
```

Works fine with DRF 3.14 but with 3.15 this now fails: `AssertionError: 'AccountSettingsView' should either include a `queryset` attribute, or override the `get_queryset()` method.`

Why is this necessary now? I don't need a whole queryset if only one single object is supported, which it gets in `get_object`, I would think? I haven't seen breaking changes about this in the changelog?

**Comment by browniebroke** [comment, marked as answer]
See this issue: https://github.com/encode/django-rest-framework/issues/9306
